### PR TITLE
#991 more robust terminal width detection for international Windows versions

### PR DESCRIPTION
**Pull request by S-Gaertner**
* changed terminal width detection regex pattern for Windows
* check for process termination in order to fail faster

### CODE REVIEW

File: src/main/java/picocli/CommandLine.java [hunk 1]
```diff
@@ -6697,7 +6699,7 @@ public void run() {
                 do {
                     if (size.intValue() >= 0) { break; }
                     try { Thread.sleep(25); } catch (InterruptedException ignored) {}
-                } while (System.currentTimeMillis() < now + 2000);
+                } while (System.currentTimeMillis() < now + 2000 && t.isAlive());
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:6702]: Looks good. Were you able to test that this prevents the 2 second timeout when the pattern cannot match anything (e.g. with the old pattern on a German language environment)?

Comment (S-Gaertner) [reply src/main/java/picocli/CommandLine.java:6702]: Yes, I manually tested that on my system with the old pattern. The 2 second delay was gone as soon as I introduced the alive test for the thread.

Comment (remkop) [reply src/main/java/picocli/CommandLine.java:6702]: Great, thank you!

**Comment by remkop** [comment]
Overall looks good. I have some other work in progress but I will try to merge this soon.

Thank you for the contribution!

**Comment by S-Gaertner** [comment]
You're welcome! Thank you for this great project :)

**Comment by remkop** [comment]
Can I ask:

* What is it that you like about picocli?
* What would you do if picocli did not exist?

**Comment by S-Gaertner** [comment]
What I like about picocli:
* It allows complex use cases, yet is pretty easy to use - especially with the convenience methods.
* It has zero dependencies - I am reluctant to include dependencies that themselves entail more dependencies.
* It has superb documentation.
* It is open source, mature, active, feature-rich and has a friendly community.

What I would do without picocli:
* I already have rolled my own command line parser for simple uses cases. Now my use cases grew more complex, so I probably would have adapted it to some degree and also simplified my use cases (and at the same time would provide more user training for the probably more complicated command line invocation).
* I also would have looked into alternatives like JCommander.

**Comment by remkop** [comment]
@S-Gaertner Great feedback, thank you! :+1:

**Pull request merged**